BLSQ/iaso
# IA-1812: Org unit form multiple states

Original issue: adding an alias cleared the org unit tree view input 
Finally this part of the code as to be totally refactored as the org unit form was using one for state and orgUniInfo too

We passed everything to Typescript in order to help with this mix-up of state (that was creating the original issue) 

Related JIRA tickets : [IA-1812](https://bluesquare.atlassian.net/browse/IA-1812)

## Self proof reading checklist

## Changes

- migrate `OrgUnitInfos` to typescript
- migrate `OrgUnitIForm` to typescript
- remove unnecessary form state in `OrgUnitInfos`
- split `OrgUnitAction` into separate file
- Wrote cypress test to test org unit creation


## How to test

Go to Orgunits > Create 
Then  select a parent orgunit and try adding an alias
Adding an alias should not empty the parent orgunit treeview

You should also try to edit and save an existing org unit

## Print screen / video

[recording-2023-01-13-16-07-51.webm](https://user-images.githubusercontent.com/25134301/212352559-52635d01-4ab8-4e75-84ec-dfbadd0e9519.webm)




[IA-1812]: https://bluesquare.atlassian.net/browse/IA-1812?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
